src/core/Akka.Remote/RemoteMetricsExtension.cs

```diff
@@ -0,0 +1,132 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="UntrustedSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Typesafe Inc. <http://www.typesafe.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using Akka.Actor;
+using Akka.Event;
+using Akka.Routing;
+
+namespace Akka.Remote
+{
+    /// <summary>
+    ///     INTERNAL API
+    ///     Extension that keeps track of remote metrics, such
+    ///     as max size of different message types.
+    /// </summary>
+    internal class RemoteMetricsExtension : ExtensionIdProvider<IRemoteMetrics>
+    {
+        public override IRemoteMetrics CreateExtension(ExtendedActorSystem system)
+        {
+            if (system.Settings.Config.GetString("akka.remote.log-frame-size-exceeding").ToLowerInvariant() == "off")
```

Comment:
yep, this is the same as the JVM - this feature is only really used for Akka.Remote / Cluster users who want to keep some historical track as to whether or not they're close to hitting the ceiling on the maximum frame size.